simtfl/node.py

```diff
@@ -56,16 +56,16 @@ def __init__(self, ident, env, network):
         Constructs a SequentialNode with the given simpy Environment and network.
```

Comment:
I don't understand the motivation for `SequentialNode`, because I assumed `simpy` provides a total ordering of events, so that the order processed by a node will be determinstic.

If my assumption is wrong, then `SequentialNode` makes sense but I would still prefer globally deterministic ordering in order to have globally deterministic simulations, unless we're absolutely certain we need to sacrifice that. (Is performance the only rationale for sacrificing globally deterministic simulation?)

Reply:
The ordering is always deterministic (if you don't write explicitly nondeterministic code). The difference between `PassiveNode` and `SequentialNode` is that `PassiveNode` doesn't block handling the next message on finishing handling this one, and `SequentialNode` does.

It's likely to be easier to write nodes with sequential message handling. However, `PassiveNode` has a simpler implementation, and concurrent message handling is sometimes necessary.

Reply:
Ah, I understand now. I missed that difference but re-reviewing `SequentialNode.run` I now see how it does not process a pending message from `self._mailbox` until all of `yield from self.handle(sender, message)` completes.